transcom/mymove
# INT B-21393 Consolidate PPM Allowable Weights

## [B-21393](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21393)

## Summary

Moves the allowable weight column from weight_tickets to ppm_shipments to that there is one consolidated allowable weight per shipment rather than one per trip.

The initial allowable weight is now calculated when the customer submits their PPM for closeout rather than in the UI as the SC opens each weight ticket.

The new allowable weight is editable by the SC when reviewing weight tickets.

This is the first time I've done a non-trivial database migration so pay extra attention to that in your review. The backfill may not be necessary. I added it to keep existing test data usable.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the customer app, create a new move with at least one PPM, submit it.
2. Login as a Services Counselor and approve the move.
3. Log back in as your original customer and upload weight tickets for at least two trips and submit the shipment for closeout. Take note of the new weight of both trips.
4. Log back in as a Services Counselor and review the documents for the move you just submitted for closeout.
5. Verify the "Shipment Info" header section is expanded upon opening the document viewer.
6. Verify the above section contains a field labeled "Allowable Weight"
7. Verify the weight ticket section below no longer contains an "Allowable Weight" section.
8. Hover over the info icon next to the allowable weight, verify a tooltip with text matching the AC appears.
9. Verify you can edit the Allowable weight in the header section.
10. Run ANDI, the new button should be focusable and have an accessible name/label
11. Approve the submitted documents.
12. Navigate to the Move history, verify the change you made to the allowable weight value is listed.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
<img width="399" alt="Screenshot 2024-10-10 at 10 11 27 AM" src="https://github.com/user-attachments/assets/ea919f06-709b-4955-9174-1c3c59944c55">
<img width="683" alt="Screenshot 2024-10-10 at 10 11 41 AM" src="https://github.com/user-attachments/assets/2fbba913-63cb-4f2d-b5d8-f493b1891b73">
<img width="1004" alt="Screenshot 2024-10-10 at 10 12 29 AM" src="https://github.com/user-attachments/assets/a0a019f2-2d3a-4105-87db-d763c7e0c228">
